ui/src/main/js/model/rest-api.js

```diff
@@ -28,21 +28,63 @@ var ajax = require('../util/ajax');
  * Workflow REST API
  */
 
+var cache = {};
+
+function addCacheEntry (key, ob) {
+    if (typeof ob === 'object') {
+        if (ob.status && (ob.status === 'SUCCESS' || ob.status === 'ABORTED' || ob.status === 'FAILED' || ob.status === 'UNSTABLE')) {
```

Comment:
@jglick We don't have a NOT_BUILT status code for this (see StatusExt).  Maybe worth caching for NOT_EXECUTED too, but at worst it's a mild performance hit for us.